packages/core/src/external/cda/process-attachments.ts

```diff
@@ -0,0 +1,219 @@
+import {
+  Attachment,
+  Bundle,
+  CodeableConcept,
+  Coding,
+  DocumentReference,
+  Extension,
+  Identifier,
+} from "@medplum/fhirtypes";
+import { executeWithNetworkRetries } from "@metriport/shared";
+import { buildDayjs } from "@metriport/shared/common/date";
+import { cloneDeep } from "lodash";
+import { MedicalDataSource, isMedicalDataSource } from "..";
+import { createAttachmentUploadPath } from "../../domain/document/upload";
+import { createFilePathFromFileName } from "../../domain/filename";
+import { executeAsynchronously } from "../../util/concurrency";
+import { Config } from "../../util/config";
+import { out } from "../../util/log";
+import { sizeInBytes } from "../../util/string";
+import { uuidv4 } from "../../util/uuid-v7";
+import { S3Utils, UploadParams } from "../aws/s3";
+import { cqExtension } from "../carequality/extension";
+import { cwExtension } from "../commonwell/extension";
+import { FhirClient } from "../fhir/api/api";
+import { convertCollectionBundleToTransactionBundle } from "../fhir/bundle/convert-to-transaction-bundle";
+import { buildDocIdFhirExtension } from "../fhir/shared/extensions/doc-id-extension";
+
+const s3Utils = new S3Utils(Config.getAWSRegion());
+const s3BucketName = Config.getMedicalDocumentsBucketName();
+
+export async function processAttachments({
+  b64Attachments,
+  cxId,
+  patientId,
+  fileName,
+  medicalDataSource,
+  fhirApi,
```

Comment:
I'm wondering if we should create the fhirApi instead of passig it in... is there a pattern for passing the fhirApi in for the core stuff or does it normally fetch it itself (maybe consolidated has a pattern to follow).

Reply:
ya, looks like we can just initialize it in the core func instead of lambda layer